tests/test_inspector.py

```diff
@@ -27,19 +28,22 @@
 from nwbinspector.utils import FilePathType, is_module_installed
 from nwbinspector.tools import make_minimal_nwbfile
 
-
-try:
-    with NWBHDF5IO(
-        path="https://dandiarchive.s3.amazonaws.com/blobs/11e/c89/11ec8933-1456-4942-922b-94e5878bb991",
-        mode="r",
-        load_namespaces=True,
-        driver="ros3",
-    ) as io:
-        nwbfile = io.read()
-    HAVE_ROS3 = True
-except ValueError:  # ValueError: h5py was built without ROS3 support, can't use ros3 driver
-    HAVE_ROS3 = False
+DANDI_TESTS_NONETWORK = os.environ.get("DANDI_TESTS_NONETWORK", "")
+NO_NETWORK = strtobool(DANDI_TESTS_NONETWORK) if DANDI_TESTS_NONETWORK != "" else False
+if not NO_NETWORK:
+    try:
+        with NWBHDF5IO(
+            path="https://dandiarchive.s3.amazonaws.com/blobs/11e/c89/11ec8933-1456-4942-922b-94e5878bb991",
+            mode="r",
+            load_namespaces=True,
+            driver="ros3",
+        ) as io:
+            nwbfile = io.read()
+        HAVE_ROS3 = True
+    except ValueError:  # ValueError: h5py was built without ROS3 support, can't use ros3 driver
+        HAVE_ROS3 = False
 HAVE_DANDI = is_module_installed("dandi")
+DISABLE_STREAMING_TESTS = NO_NETWORK or not HAVE_ROS3 or not HAVE_DANDI
```

Comment:
As far as I can tell only the ``DISABLE_STREAMING_TESTS`` variable is actually used by the tests. Instead of defining all the extra variables at the top of the module, I would suggest putting all the code to check whether streaming tests should be disabled into a function, e.g., ``check_streaming_tests_enabled()``  that you then call to set ``DISABLE_STREAMING_TESTS = check_streaming_tests_enabled()``. In this way you can also avoid duplicating the code below.